crates/compiler/parse/src/ast.rs

```diff
@@ -1746,11 +1759,9 @@ pub enum Pattern<'a> {
 
     OpaqueRef(&'a str),
 
-    Apply(
-        &'a Loc<Pattern<'a>>,
-        &'a [Loc<Pattern<'a>>],
-        PatternApplyStyle,
-    ),
+    Apply(&'a Loc<Pattern<'a>>, &'a [Loc<Pattern<'a>>]),
+
+    PncApply(&'a Loc<Pattern<'a>>, Collection<'a, Loc<Pattern<'a>>>),
```

Comment:
Why use a `Collection` instead of just a slice?

Reply:
That was the whole point of the change.  Collections handle trailing comments and have uniform formatting

Reply:
Okay, fair. We can leave this for now, and presumably in the future this won't be a problem since `Apply` is going away.